File: src/main/resources/static/app/services/category.service.js
angular.module('productosJM').service('CategoryService', function($http) {
    this.getCategories = function() {
        return $http.get('/api/category/getCategories');
    };
    this.createCategory = function(category) {
        return $http.post('/api/category/createCategory', category);
    };
    this.updateCategory = function(id, category) {
        return $http.put('/api/category/updateCategory/' + id, category);
    };
    this.deleteCategory = function(id) {
        return $http.delete('/api/category/deleteCategory/' + id);
    };
});

File: src/main/java/com/axity/test/product/services/CategoryService.java
package com.axity.test.product.services;

import com.axity.test.product.repository.CategoryRepository;
import com.axity.test.product.repository.datamodel.CategoryEntity;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/category")
public class CategoryService {
    @Autowired
    private CategoryRepository categoryRepository;

    @GetMapping("/getCategories")
    public List<CategoryEntity> getCategories() {
        return categoryRepository.findAll();
    }
    @PostMapping("/createCategory")
    public CategoryEntity addCategory(@RequestBody CategoryEntity category) {
        return categoryRepository.save(category);
    }
    @PutMapping("/updateCategory/{id_category}")
    public CategoryEntity updateCategory(@PathVariable int id_category, @RequestBody CategoryEntity category) {
        return categoryRepository.save(category);
    }

    @DeleteMapping("/deleteCategory/{id_category}")
    public void deleteCategory(@PathVariable int id_category) {
        categoryRepository.deleteById(id_category);
    }
}
